compiler/src/main/java/com/squareup/anvil/compiler/codegen/ksp/KspUtil.kt

```diff
@@ -386,11 +386,16 @@ internal fun KSType.contextualToClassName(origin: KSNode): ClassName {
 }
 
 private fun KSType.checkErrorType(origin: KSNode) {
-  if (isError) {
-    throw KspAnvilException(
-      message = "Error type '$this' is not resolvable in the current round of processing. Check your imports or, if this is a generated type, ensure the tool that generates it has its outputs appropriately sources as inputs to the KSP task.",
-      node = origin,
-    )
+  val errorType = if (this.isError) origin else arguments.find { it.type?.resolve()?.isError == true }?.type?.resolve()
+  if (errorType != null) {
+    val message = buildString {
+      append("Error type '$errorType' is not resolvable in the current round of processing. ")
+      (origin as? KSValueParameter)?.let {
+        append("This happened for parameter '${it.name?.asString()} : ${it.type.resolve()}'. ")
+      }
```

Comment:
Let's add support for `KSPropertyDeclaration` or `KSFunctionDeclaration` while we're at it. Can stick them in a `when` for the smartcast